storage/s3_test.go

```diff
@@ -128,6 +128,24 @@ func TestNewSessionWithNoSignRequest(t *testing.T) {
 	}
 }
 
+func TestNewSessionWithProfile(t *testing.T) {
+	globalSessionCache.clear()
+	profileName := "default"
```

Comment:
Thanks for warning, but to be honest I don't know how can we actually test different profiles.

profiles are read from the actual `.aws/credentials` file of the user. So the tests should not modify it.  
1) We cannot add (and remove)  temporary profiles.
    > We might not have a write access to file.
    >> Even if we have write access and add a temporary profile,  the profile name we've chosen might collide with existing profiles and we may accidentally delete actual profiles.
    >>> To avoid this we can check if there is such a profile with same name before adding temporary profile but "being able to check if there is such a profile (and use if there)" is the very thing this test was. supposed to do.
To me, it seemed as a vicious cycle and meaningless.
2) There is a high probability that  `.aws/credentials` file has a profile named `default`. So trying to obtain "default" profile it from both through our `storage.Options` and aws's `NewSharedCredentials()` seemed to a way to check if our function can actually do the same thing thing with aws.
> ps. If there is no profile with the given profileName (in this. case "default") both our code and the aws's methods give the same error.
 

ps.  by `user` I mean the person who will run this test, not necessarily the end user.

Reply:
Just to cover all the cases, AWS SDK allows specifying the credential file. So if we add this feature to `s5cmd` then we can create a [temporary file](https://pkg.go.dev/io/ioutil#TempFile) and use it to write tests that makes more senses, and also cover more cases: e.g. tests for default profile, existing profiles, non-existing profiles.